go/vt/vtgate/executor.go

```diff
@@ -202,6 +212,8 @@ func (e *Executor) execute(ctx context.Context, safeSession *SafeSession, sql st
 func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, target querypb.Target, logStats *LogStats) (*sqltypes.Result, error) {
 	if target.Shard != "" {
 		// V1 mode or V3 mode with a forced shard target
+		// TODO(sougou): change this flow to go through V3 functions
+		// which will allow us to benefit from the autocommitable flag.
```

Comment:
Even without routing through v3 it seems like we could make this change, no?

I agree that having something like PlanShardPassthrough would make sense and then we can reduce this duplication.

Reply:
`Resolver` goes through V2 code, and it should not contain any reference to autocommit. Introducing such awareness there is risk-prone.

Yeah. The better approach will be to have an opcode for this in V3. We'll still need to distinguish reads from writes because of the `canCommit` flag (to be renamed from `isFinal`)